extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -670,15 +671,20 @@ public void handle(AsyncResult<String> event) {
                 if (event.failed()) {
                     Throwable effectiveCause = event.cause();
                     if (effectiveCause instanceof BindException) {
-                        if ((sslConfig == null) && (httpServerOptions != null)) {
-                            effectiveCause = new QuarkusBindException(httpServerOptions.getPort());
-                        } else if ((httpConfiguration.insecureRequests == InsecureRequests.DISABLED) && (sslConfig != null)) {
-                            effectiveCause = new QuarkusBindException(sslConfig.getPort());
-                        } else if ((sslConfig != null) && (httpConfiguration.insecureRequests == InsecureRequests.ENABLED)
-                                && (httpServerOptions != null)) {
-                            effectiveCause = new QuarkusBindException(
-                                    List.of(httpServerOptions.getPort(), sslConfig.getPort()));
-                        }
+                        effectiveCause = new QuarkusBindException(
+                            (BindException)effectiveCause,
+                            new ArrayList<>(){{
+                                if ((sslConfig == null) && (httpServerOptions != null)) {
+                                    this.add(httpServerOptions.getPort());
+                                } else if ((httpConfiguration.insecureRequests == InsecureRequests.DISABLED) && (sslConfig != null)) {
+                                    this.add(sslConfig.getPort());
+                                } else if ((sslConfig != null) && (httpConfiguration.insecureRequests == InsecureRequests.ENABLED)
+                                           && (httpServerOptions != null)) {
+                                    this.addAll(List.of(httpServerOptions.getPort(), sslConfig.getPort()));
+                                }
```

Comment:
I really don't like this pattern. Please just build a `List` before creating your `QuarkusBindException`, it will be simpler and more readable.

Reply:
After rebasing to my local, this looks out of date. Don;t know why this was seen as latest.